src/app/pages/ConnectorsPage/components/ConnectorInstances/ConnectorInstances.tsx

```diff
@@ -111,7 +111,9 @@ export const ConnectorInstances: FunctionComponent<ConnectorInstancesProps> = ({
   );
 
   const onRemoveGroup = onClearAllFilters;
-  const isFiltered = namesChips.length > 0;
+  // isFiltered determines if the TableView component will show the "no results" block
+  // instead of the "empty state" when no items are present. See MGDCTRS-1623 for more details.
+  const isFiltered = namesChips.length > 0 || page > 1;
```

Comment:
`TableView` shared component [is handling](https://github.com/redhat-developer/app-services-ui-components/blob/028fb603423d4178c038debe237b411ce2e29fb4/src/shared/Table/TableView.tsx#L104) what to show when no items are present.
Setting the `isFiltered` flag also for page > 1 will fix the issue.